tool/autoqa/lib/canonical-extractor.js

```diff
@@ -27,10 +27,13 @@ const child_process = require('child_process');
 const EnglishLanguagePack = require('../../../lib/i18n/american-english');
 
 const VALUE_MAP = {
-    "1": "one",
-    "2": "two",
-    "3": "three",
-    "4": "four"
+    "1": ["one"],
+    "2": ["two"],
+    "3": ["three"],
+    "4": ["four"],
+    "feb 14 2017": ["february 14 2017", "february 14, 2017", "feb 14, 2017", "14 february 2017", "14 february, 2017", "14 feb 2017", "14 feb, 2017"],
+    "may 4th, 2016": ["may 4 2016", "may 4, 2016", "may 4th 2016", "4th may 2016", "4th may, 2016", "4 may 2016", "4 may, 2016"],
+    "august 2nd 2017": ["august 2, 2017", "august 2 2017", "august 2nd, 2017", "2 august 2017", "2nd august 2017", "2 august, 2017", "2nd august, 2017"]
```

Comment:
Is it worth using the tokenizer to extract these? We can already identify dates pretty well.

Reply:
Yeah, the extractor will be updated soon with an actual parser. I will do a separate PR to clean this up altogether.